### PR TITLE
Count

### PR DESCRIPTION
**Pull request by Muhammadrasul446**
### **Describe your change:**
Added counting islands in a matrix

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: matrix/count_islands_in_matrix.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+class matrix: # Public class to implement a graph
+ 
+	def __init__(self, row, col, g): # Class constructor
+		self.ROW = row 
+		self.COL = col 
+		self.graph = g
+
+	def is_safe(self, i, j, visited): 
+		return (i >= 0 and i < self.ROW and j >= 0 and j < self.COL and not visited[i][j] and self.graph[i][j])
```
Comment (cclauss) [inline matrix/count_islands_in_matrix.py:12]:
```suggestion
		return 0 <= i < self.ROW and 0 <= j < self.COL and not visited[i][j] and self.graph[i][j]
```

File: matrix/count_islands_in_matrix.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+			if self.is_safe(i + rowNbr[k], j + colNbr[k], visited): 
+				self.diffs(i + rowNbr[k], j + colNbr[k], visited) 
+ 
+ 
+	def count_islands(self): # And finally, count all islandas.
+		visited = [[False for j in range(self.COL)]for i in range(self.ROW)] 
+		count = 0
+		for i in range(self.ROW): 
+			for j in range(self.COL): 
+				if visited[i][j] == False and self.graph[i][j] == 1:
```
Comment (cclauss) [inline matrix/count_islands_in_matrix.py:31]:
```suggestion
				if visited[i][j] is False and self.graph[i][j] == 1: 
```

File: matrix/count_islands_in_matrix.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# An island in matrix is a group of linked areas, all having the same value. 
+# This code couts number of islands in a given matrix, with including diagonal connections.
+
+class matrix: # Public class to implement a graph
+ 
+	def __init__(self, row, col, g): # Class constructor
```
Comment (cclauss) [inline matrix/count_islands_in_matrix.py:6]:
```suggestion
	def __init__(self, row: int, col: int, graph: list): # Class constructor
```
Python type hints as discussed in CONTRIBUTING.md.

**Comment by Muhammadrasul446** [comment]
Committed all suggestions

**Comment by Muhammadrasul446** [comment]
> Hey @Muhammadrasul446,
> Something went wrong with the build.
> 
> TravisCI finished with status `errored`, which means the build failed because of something unrelated to the tests, such as a problem with a dependency or the build process itself.
> 
> [View build log](https://travis-ci.com/TheAlgorithms/Python/builds/170329869)
> ###### TravisBuddy Request Identifier: 772eb870-a9a5-11ea-9161-4f77e253115e

Hey there. I am knew to github, so should i commit and then create a new pull request or what? I've seen the build log.

**Comment by cclauss** [comment]
This is looking good.  Tab indentation will fail our tests but I think that I fixed that.  Can you please provides some sample data that proves that this algorithm does what it says that it does.  One list that should contain a several islands and another list that should contain no islands.  Thanks.

**Pull request merged**